addon/webextension/background/main.js

```diff
@@ -130,19 +130,26 @@ this.main = (function() {
   }
 
   exports.onClickedContextMenu = catcher.watchFunction((info, tab) => {
-    if (!tab) {
-      // Not in a page/tab context, ignore
-      return;
-    }
-    if (!urlEnabled(tab.url)) {
-      senderror.showError({
-        popupMessage: "UNSHOOTABLE_PAGE"
-      });
-      return;
-    }
-    catcher.watchPromise(
-      toggleSelector(tab)
-        .then(() => sendEvent("start-shot", "context-menu", {incognito: tab.incognito})));
+    catcher.watchPromise(hasSeenOnboarding.then(onboarded => {
```

Comment:
I think the flow here should be something like:

```js
hasSeenOnboarding.then(onboarded => {
  if (!onboarded) { ... }
  if (!tab) { ... }
  if (!urlEnabled) { ... }
  // then, toggleSelector()...
})
```

This way, no matter how you first trigger screenshots, you'll see onboarding as a new user.

Reply:
Good point! Thanks for continuing to work on this, I've gotten a bit behind on reviews. I'll take a look today or Monday :-)